### PR TITLE
Find modules for MHProxies.asInterfaceInstance's hidden classes

### PR DESCRIPTION
**Pull request by babsingh**
In JDK22, `MethodHandleProxies.asInterfaceInstance` was reimplemented
using a more direct `MethodHandle` approach to improve performance.
This approach introduces hidden classes which have an interface for
its host class. A dynamic module is created for each such hidden
class. This PR correctly finds and sets `java.lang.Class.module` for
such hidden classes.

Fixes: #18718

### CODE REVIEW

**Comment by babsingh** [comment]
@hangshao0 Can you please review this PR?

File: runtime/vm/createramclass.cpp [hunk 1]
```diff
@@ -3655,16 +3669,17 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
...
 						module = javaVM->unamedModuleForSystemLoader;
 					}
 				}
+				if (findModule) {
+					U_32 pkgNameLength = (U_32)packageNameLength(romClass);
+					omrthread_monitor_t classLoaderModuleAndLocationMutex = javaVM->classLoaderModuleAndLocationMutex;
+					omrthread_monitor_enter(classLoaderModuleAndLocationMutex);
+					module = findModuleForPackage(vmThread, findModuleClassLoader, J9UTF8_DATA(className), pkgNameLength);
+					omrthread_monitor_exit(classLoaderModuleAndLocationMutex);
+				}
```
Comment (hangshao0) [inline runtime/vm/createramclass.cpp:3682]: An assertion could be added when `findModule` is false: 
`Assert_VM_true((module == javaVM->unamedModuleForSystemLoader) || (module == hostClass->module))`.

Comment (babsingh) [reply runtime/vm/createramclass.cpp:3684]: Assertion has been added.

**Comment by hangshao0** [comment]
jenkins test sanity.functional,extended.functional alinux jdk22

**Comment by hangshao0** [comment]
jenkins test sanity.openjdk,extended.openjdk alinux jdk22

**Comment by babsingh** [comment]
> Are you looking at the failures in the PR builds ?

Yes, I am opening new issues for those failures. They are unrelated. I will post a comment in this PR once I am done.

**Comment by babsingh** [comment]
Known JDK22 failures:
- https://github.com/eclipse-openj9/openj9/issues/18696
- https://github.com/eclipse-openj9/openj9/issues/18697
- https://github.com/eclipse-openj9/openj9/issues/18703
- https://github.com/eclipse-openj9/openj9/issues/18693
- https://github.com/eclipse-openj9/openj9/issues/18763

New JDK22 failures (fyi @pshipton, @JasonFengJ9, @tajila):
- https://github.com/eclipse-openj9/openj9/issues/18804
- https://github.com/eclipse-openj9/openj9/issues/18805
- https://github.com/eclipse-openj9/openj9/issues/18807
- https://github.com/eclipse-openj9/openj9/issues/18809
- https://github.com/eclipse-openj9/openj9/issues/18810
- https://github.com/eclipse-openj9/openj9/issues/18811
- https://github.com/eclipse-openj9/openj9/issues/18812

@hangshao0 My changes are not responsible for the PR build failures. Some failures are known, and new issues have been opened for the failures, which are seen for the first time.

**Review by hangshao0 — APPROVED** [review]
LGTM.

**Pull request merged**